crates/edr_napi_core/src/logger.rs

```diff
@@ -0,0 +1,1193 @@
+use core::fmt::Debug;
+use std::{fmt::Display, marker::PhantomData, sync::Arc};
+
+use ansi_term::{Color, Style};
+use derive_where::derive_where;
+use edr_eth::{result::ExecutionResult, transaction::ExecutableTransaction, Bytes, B256, U256};
+use edr_evm::{
+    blockchain::BlockchainError,
+    precompile::{self, Precompiles},
+    trace::{AfterMessage, Trace, TraceMessage},
+    transaction::Transaction as _,
+    SyncBlock,
+};
+use edr_provider::{
+    time::CurrentTime, CallResult, DebugMineBlockResult, EstimateGasFailure, ProviderError,
+    ProviderSpec, TransactionFailure,
+};
+use itertools::izip;
+
+/// Trait for a function that decodes console log inputs.
+pub trait DecodeConsoleLogInputsFn: Fn(Vec<Bytes>) -> Vec<String> + Send + Sync {}
```

Comment:
We now deal with Rust `Fn`s instead of the `ThreadSafeFunction` and `JsFunction` in this file. This creates a clear demarcation between marshalled types and internal types.